File: test/memory_advice/testapp/app/src/main/java/com/memory_advice/testapp/MainActivity.java
package com.memory_advice.testapp;

import android.text.method.ScrollingMovementMethod;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

  // Used to load the 'testapp' library on application startup.
  static {
    System.loadLibrary("native-lib");
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    TextView tv = findViewById(R.id.sample_text);
    tv.setMovementMethod(new ScrollingMovementMethod());
    tv.setText(runTests());
  }

  public native String runTests();
}

File: test/tuningfork/metrics/app/src/main/java/test/tuningfork/metricstest/MainActivity.kt
package test.tuningfork.metricstest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Debug
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        info_text.text = measureMetricsPerformance(Debug::class.java)
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    external fun measureMetricsPerformance(d: Class<Debug>): String

    companion object {
        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("native-lib")
        }
    }
}
